validator/tests/unit3/test_context_manager/tests.py

```diff
@@ -15,15 +15,163 @@
 
 import unittest
 
+from collections import namedtuple
+import hashlib
+import time
+
 from sawtooth_validator.database import dict_database
 from sawtooth_validator.execution import context_manager
 from sawtooth_validator.state.merkle import MerkleDatabase
 from sawtooth_validator.state.state_delta_store import StateDeltaStore
 from sawtooth_validator.protobuf.state_delta_pb2 import StateChange
 
 
+TestAddresses = namedtuple('TestAddresses',
+                              ['inputs', 'outputs', 'reads', 'writes'])
+
+
 class TestContextManager(unittest.TestCase):
 
+    def setUp(self):
+        self.database_of_record = dict_database.DictDatabase()
+        self.state_delta_store = StateDeltaStore(dict_database.DictDatabase())
+        self.context_manager = context_manager.ContextManager(
+            self.database_of_record, self.state_delta_store)
+        self.first_state_hash = self.context_manager.get_first_root()
+
+        # used for replicating state hash through direct merkle tree updates
+        self.database_results = dict_database.DictDatabase()
+
+    def tearDown(self):
+        self.context_manager.stop()
+
+    def _setup_context(self):
+        # 1) Create transaction data
+        first_transaction = {'inputs': ['aaaa', 'bbbb', 'cccc'],
+                             'outputs': ['llaa', 'aall', 'nnnn']}
```

Comment:
It might be nice to clean up these addresses now, since you are working this sprint on enforcing address characteristics. I'll update the PR to only use valid addresses.